cms/templates/js/staff-lock-editor.underscore

```diff
@@ -29,4 +29,46 @@
         </li>
     </ul>
 </div>
+<br>
+
+<% if (xblockInfo.isVertical()) { %>
+    <h3 class="modal-section-title">
+        <%- gettext('Unit Access') %>
+    </h3>
+    <div class="modal-section-content access-change">
+        <ul class="list-fields list-input">
+            <li>
+            <%- gettext('Restrict access to:') %>
+            <select class="user-partition-select">
+                <option value="none">Choose one</option>
+                <% for (var i=0; i < contentGroups.length; i++) { %>
+                    <option id="<%= contentGroups[i].id %>-option" value="<%= contentGroups[i].id %>"><%= contentGroups[i].name %></option>
+                <% } %>
+            </select>
+            </li>
+            <br>
+            <% if (contentGroups.length > 0) { %>
```

Comment:
Note that you will have to deal with deleted groups, in the same way that visibility_editor.html and its associated code does.

Also don't forget i18n and safe template practices. 
http://edx.readthedocs.io/projects/edx-developer-guide/en/latest/conventions/internationalization/i18n.html

http://edx.readthedocs.io/projects/edx-developer-guide/en/latest/conventions/safe_templates.html